File: Programmers/java/LEVEL_2_연속_부분_수열_합의_개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        Map<Integer, Set<Integer>> map = new HashMap<>();
        Set<Integer> removeDup = new HashSet<>();

        for (int i = 0; i < elements.length; i++) {
            int sum = 0, cnt = 1;

            for (int j = i; j < elements.length + i; j++) {
                int idx = j;

                if (idx >= elements.length) {
                    idx = idx - elements.length;
                }

                sum += elements[idx];

                if (!map.containsKey(cnt)) {
                    map.put(cnt, new TreeSet<>());
                }
                map.get(cnt).add(sum);
                cnt++;
            }
        }

        for (int key : map.keySet()) {
            for (int n : map.get(key)) {
                removeDup.add(n);
            }
        }

        answer = removeDup.size();

        return answer;
    }
}

File: Programmers/cpp/LEVEL_2_연속_부분_수열_합의_개수.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <set>

using namespace std;

int solution(vector<int> elements) {
	int answer = 0;
	unordered_map<int, set<int>> um;
	set<int> removeDup;

	for (int i = 0; i < elements.size(); i++) {
		int sum = 0, cnt = 1;

		for (int j = i; j < elements.size() + i; j++) {
			int idx = j;

			if (idx >= elements.size()) {
				idx = idx - elements.size();
			}

			sum += elements[idx];
			um[cnt].insert(sum);
			cnt++;
		}
	}

	for (pair<int, set<int>> m : um) {
		for (int n : m.second) {
			removeDup.insert(n);
		}
	}

	answer = removeDup.size();

	return answer;
}